apps/web/src/components/activity/ExecutionDetailsSteps.tsx

```diff
@@ -0,0 +1,287 @@
+import { Grid, Text } from '@mantine/core';
+import styled from 'styled-components';
+import { format, isValid, parseISO } from 'date-fns';
+import { ExecutionDetailsStatusEnum } from '@novu/shared';
+
+import { colors, shadows } from '../../design-system';
+import { Chat, CheckCircle, InApp, Mail, Sms, Timer } from '../../design-system/icons';
+
+const ExecutionDetailsStepItem = styled.div`
+  padding: 20px;
+  padding-left: 15px;
+  border: 1px solid ${colors.B30};
+  margin-bottom: 15px;
+  border-radius: 7px;
+  color: ${colors.B80};
+`;
+
+const ExecutionDetailsStepFailed = styled(ExecutionDetailsStepItem)`
+  background: rgba(230, 69, 69, 0.15);
+  border: 1px solid ${colors.error};
+`;
+
+const ExecutionDetailsStepSuccess = styled(ExecutionDetailsStepItem)`
+  background: rgba(77, 153, 128, 0.15);
+  border: 1px solid ${colors.success};
+`;
+
+const StepTimeFlowWrapper = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  margin: 0 5px;
+  padding: 0;
+  &:last-child {
+    margin: 0 0 0 5px;
+  }
+`;
+
+const StepTimeWrapper = styled(StepTimeFlowWrapper)`
+  flex-flow: column;
+`;
+
+interface ISeparator {
+  show: boolean;
+}
+
+/**
+ * TODO: It should be an icon for an arrow but I haven't found it
+ * so doing it CSS wise
+ */
+const Separator = styled.div<ISeparator>`
+  border: 1px dashed ${colors.white};
+  border-radius: 10px;
+  height: 1px;
+  opacity: 0.8;
+  position: relative;
+  top: 13px;
+  visibility: ${(props) => (props.show ? 'visible' : 'hidden')};
+  width: 30px;
+`;
+
+const StepName = styled.div`
+  color: ${colors.white};
```

Comment:
Don't forget to factor in the color scheme change. This will be a problem with light color scheme...
I suggest using Text and Title from our design system, it does all of those for you(also makes sure of font) and it helps us be consistent throughout the project.

Reply:
How do you mark other colours non theme based for secondary items such gray texts? Does the theme and the design system components take care of that?

Reply:
I'm not sure I understand what you mean. But you can use <Text color={} size={} .../>.